packages/tax/src/services/tax-module-service.ts

```diff
@@ -166,4 +167,145 @@ export default class TaxModuleService<
   ) {
     return await this.taxRateRuleService_.create(data, sharedContext)
   }
+
+  async getTaxLines(
+    items: (TaxTypes.TaxableItemDTO | TaxTypes.TaxableShippingDTO)[],
+    calculationContext: TaxTypes.TaxCalculationContext,
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<(TaxTypes.ItemTaxLineDTO | TaxTypes.ShippingTaxLineDTO)[]> {
+    const regions = await this.taxRegionService_.list(
+      {
+        $or: [
+          {
+            country_code: calculationContext.address.country_code,
+            province_code: null,
+          },
+          {
+            country_code: calculationContext.address.country_code,
+            province_code: calculationContext.address.province_code,
+          },
+        ],
+      },
+      {},
+      sharedContext
+    )
+
+    const toReturn = await promiseAll(
+      items.map(async (item) => {
+        const isShipping = "shipping_option_id" in item
+        let ruleQuery = isShipping
```

Comment:
We have quite a bit of nesting here, I'd suggest we break this method down with a couple of helper methods (eg. `getRulesForItem` or `getRatesForItem`, and the rest can stay in-line)

Reply:
Actually looking at the code below as well, a better approach might be to simply do:
```
if (isShipping)
 return getShippingRate

return productRate
```

And then the two calculation are independent. It will result in a bit of code duplication, but you will have far less conditionals, so it will be much easier to read and follow.